src/LinalgMsg.chpl

```diff
@@ -392,9 +351,41 @@ module LinalgMsg {
             cmd,dtype2str(x1G.dtype),dtype2str(x2G.dtype),rname));
 
         // assumes both arrays have been broadcasted to ND dimensions
-        proc doVecdot(type x1Type, type x2Type, type resultType): MsgTuple throws {
-            var ex1 = toSymEntry(x1G, x1Type, nd),
-                ex2 = toSymEntry(x2G, x2Type, nd);
+
+        // See the comment in matmul for a description of how this deduces the output type
+        // from the input types.  The one exception here is that "both inputs are bool" is
+        // not allowed.
+
+        select (x1G.dtype, x2G.dtype) {
+        // real cases
+            when (DType.Float64, DType.Int64)   do return doVecdot(real,    int,     real);
+            when (DType.Float64, DType.UInt8)   do return doVecdot(real,    uint(8), real);
+            when (DType.Float64, DType.Float64) do return doVecdot(real,    real,    real);
+            when (DType.Float64, DType.Bool)    do return doVecdot(real,    bool,    real);
+            when (DType.Int64, DType.Float64)   do return doVecdot(int,     real,    real);
+            when (DType.UInt8, DType.Float64)   do return doVecdot(uint(8), real,    real);
```

Comment:
we don't have to make this change, but is this the same use case for `compute_result_type` as matrix multiplication

Reply:
It's almost identical, except that vecdot doesn't allow both inputs to be bool.

Reply:
OK, I refactored this one too.